packages/vscode-ui5-language-assistant/.vscodeignore

```diff
@@ -3,20 +3,23 @@
 node_modules
 
 # Except a **subset** language-server package which is executed as a separate process
-# Note only one star (*) is expended in these dirs.
-!**/@ui5-language-assistant/language-server/*
-!**/@ui5-language-assistant/language-server/dist/*
-
-# For debugging of bundled language-server in production (source maps)
-!**/@ui5-language-assistant/language-server/lib/src/**
-!**/@ui5-language-assistant/language-server/src/**
+! **/@ui5-language-assistant/language-server/{*.d.ts,CHANGELOG.md,CONTRIBUTING.md,README.md,package.json}
+! **/@ui5-language-assistant/language-server/dist/**/*.js
+! **/@ui5-language-assistant/language-server/lib/src/**/*.{js,d.ts,json}
 
 # The VSCode Extension's dev artifacts
-coverage/**
-.nyc_output/**
-lib/test/**
-test/**
-scripts
+# Exclude all
+**/*
+
+# Except mandatory set which should be included in the vsix
+! CHANGELOG.md
+! CONTRIBUTING.md
+! LICENSE
+! README.md
+! LICENSES/*
+! package.json
 
-# Readme gifs (linked via jsdeliver instead of part of vsix).
-resources/readme
\ No newline at end of file
+! dist/**/*.js
+! lib/src/**/*.{js,d.ts,json}
```

Comment:
if you only need runtime artifacts you don't need the `d.ts` they are design time artifacts